language-design.md

```diff
@@ -28,7 +38,30 @@ you know a poorly designed language? What are the symptoms?
+
+
+Interestingly, in the article about Quorum, Pavlus makes the implicit claim that well-designed languages should have very low learning curves.  This is not a statement that we agreed with, as there needs to be a balance between ease of use and long term productivity (more on this later).
+
+
+A poorly designed language is one that does not design for growth.  A symptom would be that the language requires intensive and very low-level hacking in order to make changes to the language that look natural.  Steele uses APL as an example, complaining that there was “no way for a user to grow the language in a smooth way” [Steele, 1998, p 6]. 
+
+
+
+Joshua:
 
+There are a lot of characteristics to a well-designed language. Although, I think pretty much all of them can be boiled down to one thing: productivity. Many of the properties of programming languages (or lack thereof) that Guy Steele supports would enhance user productivity, including starting them small and designing them to be able to be grown by their users. However, this may be quite unintuitive as many people would suspect that it would be easier to get more things done if you spent less time adding features. However, when a languages has too many features built in, it can become overwhelming to understand. Modern versions of C++ have become this way to some degree. There are so many advanced features now that C++ can do pretty much anything. Some of it looks like Python, some of it looks like C, and some of it is hard to decipher without already being familiar whith what the code is supposed to do. This actually hinders productivity because it becomes harder to maintain code as users are free to implement their programs in a wide variety of ways. One solution to this is to move functionality out of the core language and into standard libraries that are easily accessible. This way users can still access these features if they need them, but there is almost an implicit discouragement to do so if the libraries are not necessary. While C++ is pretty good about this, it can make it hard to include all but the simplest of external libraries. This has encouraged people to put all of their library code into a single, easy-to-include file rather than breaking it down into files and folders like they normally might. The ability to grow a language also contributes to productivity. By making it easy to extend a language's core functionality, programmers can better integrate libraries into their code and take advantage of the features they bring without having to sacrifice readability. Other features that Guy Steele mentioned, like generics and method overloading, also help to simplify and speed up the process of integrating more functionality into the language, therefore increasing programmer productivity.
```

Comment:
I think you're right, and I really like your point about how language design should be productivity-driven. but how do you define productivity? Does that mean getting as much done as possible in the shortest time? Or does it perhaps mean running code as efficiently as possible? It could also mean other things, like giving the programmer the largest amount of flexibility to express themselves/manipulate the computer. 
If productivity simply means "getting out of the way of the programmer and making it easy for them to code", then I think you're making a lot of the same points as Pavlus, in that a language should first and foremost be easy to learn.